server/controller/profile.js

```diff
@@ -0,0 +1,76 @@
+import models from '../models';
+
+const { User, Profile } = models;
+/**
+ * A class that handles user profile operations
+* */
+class ProfileController {
+  /**
+   * create a profile
+   * @param {object} req - request object
+   * @param {object} res - response object
+   * @param{function} next - next function
+   * @returns {object} response object
+   *
+   */
+  static async createProfile(req, res, next) {
+    const { userId } = req.user;
+
+    try {
+      const user = await User.findByPk(userId);
+      if (!user) {
+        return res.status(404).json({
+          status: 404,
+          error: 'This user does not exist',
+        });
+      }
+
+      const {
+        age, gender, phoneNumber, isEmployed, bio, yrsOfExperience, industry, image,
+        isEnrolled, progress, isCertified, instagram, twitter, facebook, linkedIn
+      } = req.body;
+
+      const [newProfile, created] = await Profile.findOrCreate({
+        where: { userId },
+        defaults: {
+          age,
+          phoneNumber,
+          gender,
+          userId,
+          isEmployed,
+          bio,
+          yrsOfExperience,
+          industry,
+          image,
+          isEnrolled,
+          progress,
+          isCertified,
+          instagram,
+          twitter,
+          facebook,
+          linkedIn
+        }
+      });
+
+      if (!created) {
```

Comment:
Hi Man, I was thinking if a profile already exists, the information they submit should update the profile instead of returning user profile already exist.

I want to assume one endpoint handles profile creation and updating.

Reply:
Actually I didn't implement it that way, this route was to be *just* for creating of profiles.
There was going to be another separate route for updating profiles. But I could refactor this to handle both.